File: t/CS/PDFFormsDemo/PDFForms/PDFFormHelpers/PDFFormControlsPropertiesHelper.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Leadtools;
using Leadtools.Pdf;


namespace PDFFormsDemo
{
   public class PDFFormControlsPropertiesHelper
   {
      public static double PDFDocumentDefaultResolution = 72.0;

      public static void SetControlBounds(FormFieldControl fieldControl, PDFRect bounds, int docResolution)
      {
         double pdfResolutionFactor = docResolution / 72.0;

         LeadRect rect = new LeadRect(
                     (int)(bounds.Left * pdfResolutionFactor),
                     (int)(bounds.Top * pdfResolutionFactor),
                     (int)(bounds.Width * pdfResolutionFactor),
                     (int)(bounds.Height * pdfResolutionFactor));

         fieldControl.FiedlBounds = rect;
      }

      public static void SetControlFont(FormFieldControl fieldControl, string fontName, int fontSize, RasterColor textColor)
      {
         fontName = string.IsNullOrEmpty(fontName) ? "Arial" : fontName;

         fieldControl.AutoFontResize = (fontSize == -1 || fontSize == 0);

         fieldControl.Font = new Font(fontName, fieldControl.AutoFontResize ? FormFieldControl.MaxFontSize : fontSize);

         fieldControl.ForeColor = Color.FromArgb(textColor.A, textColor.R, textColor.G, textColor.B);
      }

      public static void SetControlFlagsProperties(FormFieldControl fieldControl, long viewFlags, long fieldFlags)
      {
         bool Hidden = (viewFlags & PDFFormField.ViewFlagsHidden) == PDFFormField.ViewFlagsHidden;
         bool NoView = (viewFlags & PDFFormField.ViewFlagsNoView) == PDFFormField.ViewFlagsNoView;
         bool Print = (viewFlags & PDFFormField.ViewFlagsPrint) == PDFFormField.ViewFlagsPrint;
         bool Locked = (viewFlags & PDFFormField.ViewFlagsLocked) == PDFFormField.ViewFlagsLocked;
         bool NoRotate = (viewFlags & PDFFormField.ViewFlagsNoRotate) == PDFFormField.ViewFlagsNoRotate;
         bool ReadOnly = (fieldFlags & PDFFormField.FieldFlagsReadOnly) == PDFFormField.FieldFlagsReadOnly;

         if (Hidden)
         {
            fieldControl.IsFieldVisible = false;
            fieldControl.IsFieldPrintable = false;
         }
         else
         {
            if (NoView)
               fieldControl.IsFieldVisible = false;
            else
               fieldControl.IsFieldVisible = true;

            if (Print)
               fieldControl.IsFieldPrintable = true;
            else
               fieldControl.IsFieldPrintable = false;
         }

         if (Locked)
            fieldControl.IsFieldLocked = true;
         else
            fieldControl.IsFieldLocked = false;

         if (ReadOnly)
            fieldControl.IsReadOnly = true;
         else
            fieldControl.IsReadOnly = false;

         if (NoRotate)
            fieldControl.IsFieldRotatable = false;
         else
            fieldControl.IsFieldRotatable = true;
      }

      public static void SetControlBorder(FormFieldControl fieldControl, RasterColor borderColor, int borderWidth, int borderStyle)
      {
         fieldControl.BorderThickness = borderWidth;
         fieldControl.BorderColor = Color.FromArgb(borderColor.A, borderColor.R, borderColor.G, borderColor.B);

         switch (borderStyle)
         {
            case PDFFormField.BorderStyleSolid:
               fieldControl.FieldBorderStyle = FieldBorderStyle.Solid;
               break;

            case PDFFormField.BorderStyleDashed:
               fieldControl.FieldBorderStyle = FieldBorderStyle.Dashed;
               break;

            case PDFFormField.BorderStyleBeveled:
               fieldControl.FieldBorderStyle = FieldBorderStyle.Beveled;
               break;

            case PDFFormField.BorderStyleInset:
               fieldControl.FieldBorderStyle = FieldBorderStyle.Inset;
               break;

            case PDFFormField.BorderStyleUnderline:
               fieldControl.FieldBorderStyle = FieldBorderStyle.Underlined;
               break;
         }
      }

      public static void SetControlBackground(FormFieldControl fieldControl, RasterColor fillColor, int fillMode)
      {
         if (fillMode == PDFFormField.FillModeFilled)
         {
            fieldControl.BackgroundColor = Color.FromArgb(fillColor.A, fillColor.R, fillColor.G, fillColor.B);
         }
         else
         {
            fieldControl.BackgroundColor = Color.White;
         }

         fieldControl.BackColor = Color.White;
      }
   }
}

File: t/VB/PDFFormsDemo/PDFForms/PDFFormHelpers/PDFFormControlsPropertiesHelper.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************

Imports System.Collections.Generic
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Imports Leadtools
Imports Leadtools.Pdf


Namespace PDFFormsDemo
   Public Class PDFFormControlsPropertiesHelper
      Public Shared PDFDocumentDefaultResolution As Double = 72.0

      Public Shared Sub SetControlBounds(fieldControl As FormFieldControl, bounds As PDFRect, docResolution As Integer)
         Dim pdfResolutionFactor As Double = docResolution / 72.0

         Dim rect As New LeadRect(CInt(bounds.Left * pdfResolutionFactor), CInt(bounds.Top * pdfResolutionFactor), CInt(bounds.Width * pdfResolutionFactor), CInt(bounds.Height * pdfResolutionFactor))

         fieldControl.FiedlBounds = rect
      End Sub

      Public Shared Sub SetControlFont(fieldControl As FormFieldControl, fontName As String, fontSize As Integer, textColor As RasterColor)
         fontName = If(String.IsNullOrEmpty(fontName), "Arial", fontName)

         fieldControl.AutoFontResize = (fontSize = -1 OrElse fontSize = 0)

         fieldControl.Font = New Font(fontName, If(fieldControl.AutoFontResize, FormFieldControl.MaxFontSize, fontSize))

         fieldControl.ForeColor = Color.FromArgb(textColor.A, textColor.R, textColor.G, textColor.B)
      End Sub

      Public Shared Sub SetControlFlagsProperties(fieldControl As FormFieldControl, viewFlags As Long, fieldFlags As Long)
         Dim Hidden As Boolean = (viewFlags And PDFFormField.ViewFlagsHidden) = PDFFormField.ViewFlagsHidden
         Dim NoView As Boolean = (viewFlags And PDFFormField.ViewFlagsNoView) = PDFFormField.ViewFlagsNoView
         Dim Print As Boolean = (viewFlags And PDFFormField.ViewFlagsPrint) = PDFFormField.ViewFlagsPrint
         Dim Locked As Boolean = (viewFlags And PDFFormField.ViewFlagsLocked) = PDFFormField.ViewFlagsLocked
         Dim NoRotate As Boolean = (viewFlags And PDFFormField.ViewFlagsNoRotate) = PDFFormField.ViewFlagsNoRotate
         Dim [ReadOnly] As Boolean = (fieldFlags And PDFFormField.FieldFlagsReadOnly) = PDFFormField.FieldFlagsReadOnly

         If Hidden Then
            fieldControl.IsFieldVisible = False
            fieldControl.IsFieldPrintable = False
         Else
            If NoView Then
               fieldControl.IsFieldVisible = False
            Else
               fieldControl.IsFieldVisible = True
            End If

            If Print Then
               fieldControl.IsFieldPrintable = True
            Else
               fieldControl.IsFieldPrintable = False
            End If
         End If

         If Locked Then
            fieldControl.IsFieldLocked = True
         Else
            fieldControl.IsFieldLocked = False
         End If

         If [ReadOnly] Then
            fieldControl.IsReadOnly = True
         Else
            fieldControl.IsReadOnly = False
         End If

         If NoRotate Then
            fieldControl.IsFieldRotatable = False
         Else
            fieldControl.IsFieldRotatable = True
         End If
      End Sub

      Public Shared Sub SetControlBorder(fieldControl As FormFieldControl, borderColor As RasterColor, borderWidth As Integer, borderStyle As Integer)
         fieldControl.BorderThickness = borderWidth
         fieldControl.BorderColor = Color.FromArgb(borderColor.A, borderColor.R, borderColor.G, borderColor.B)

         Select Case borderStyle
            Case PDFFormField.BorderStyleSolid
               fieldControl.FieldBorderStyle = FieldBorderStyle.Solid
               Exit Select

            Case PDFFormField.BorderStyleDashed
               fieldControl.FieldBorderStyle = FieldBorderStyle.Dashed
               Exit Select

            Case PDFFormField.BorderStyleBeveled
               fieldControl.FieldBorderStyle = FieldBorderStyle.Beveled
               Exit Select

            Case PDFFormField.BorderStyleInset
               fieldControl.FieldBorderStyle = FieldBorderStyle.Inset
               Exit Select

            Case PDFFormField.BorderStyleUnderline
               fieldControl.FieldBorderStyle = FieldBorderStyle.Underlined
               Exit Select
         End Select
      End Sub

      Public Shared Sub SetControlBackground(fieldControl As FormFieldControl, fillColor As RasterColor, fillMode As Integer)
         If fillMode = PDFFormField.FillModeFilled Then
            fieldControl.BackgroundColor = Color.FromArgb(fillColor.A, fillColor.R, fillColor.G, fillColor.B)
         Else
            fieldControl.BackgroundColor = Color.White
         End If

         fieldControl.BackColor = Color.White
      End Sub
   End Class
End Namespace
